apps/homescreen/js/datastore.js

```diff
@@ -198,7 +201,9 @@
           (event) => {
             var cursor = event.target.result;
             if (cursor) {
-              results.push(cursor.value);
+              if (filter(cursor.value)) {
```

Comment:
This is for only storing/displaying pinned pages from the places datastore (which has all history) - as far as I know, there's no way of creating an index/query directly from a DataStore: https://developer.mozilla.org/en-US/docs/Web/API/DataStore (the spec even says to do what we're doing - synchronise and keep your own index for querying purposes)